### DISCUSSION TITLE
IK Sync for OSC Parmeters

### DISCUSSION THREAD

**Discussion by Dervali-git** [vrchat-community/osc#9] [Ideas]
Would it be possible for OSC Parameters to use IK Sync speed instead of Playable? This would greatly improve the appearance of certain interrogations that may require a quick refresh rate such as face or eye tracking. I could see this accomplished a couple ways, either an optional command that gets sent via OSC telling VRC to switch the sync type, or have it auto switch if an OSC command was received for that parameter in the last X seconds. 

Parameters already can use IK sync in puppets due to the obvious need for those actions to happen in real time `When a Expression parameter is in-use in a Puppet menu, it automatically swaps from Playable to IK sync so you get the faster sync.` so it would make extend this to OSC. 

For reference [Sync Types](https://docs.vrchat.com/docs/animator-parameters#sync-types)

**Comment by knah** [comment]
This probably would be best controllable from the external application. Some things would benefit greatly from IK sync, like face tracking, whereas for some other things, like heart rate monitoring, it won't provide much value and they can be left with playable sync just fine.

**Reply by Shadowriver** [reply]
Or make it as a parameter configuration
